pubsub/iterator.go

```diff
@@ -309,6 +328,24 @@ func (it *messageIterator) receive(maxToPull int32) ([]*Message, error) {
 				pendingMessages[ackID] = m
 			}
 		}
+
+		if it.enableTracing {
+			ctx := context.Background()
+			if m.Attributes != nil {
+				ctx = propagation.TraceContext{}.Extract(ctx, newMessageCarrier(m))
+			}
+			attr := getSubscriberOpts(it.projectID, it.subID, m)
+			_, span := startSpan(ctx, subscribeSpanName, it.subID, attr...)
```

Comment:
The custom `startSpan` just alleviates the need to call `fmt.Sprintf` for a few spans (create, subscribe, process) since those need to include the topic/subscriptionID. It's probably not necessary for all spans, and I could just call tracer().Start for most spans if necessary.

The extraction only happens once since this is the only place where the process boundary is crossed (publish ->subscribe). Do you think it's worth creating a custom utility function for this purpose? I suppose it would make the testing easier.

I don't use the new context since I don't want the parent/child relationship in most places.